File: src/android/src/com/christian/sentimask/SMProcessingDetails.java
package com.christian.sentimask;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import com.christian.sentimask.SMBase;
import com.christian.sentimask.SMPointF;
import com.christian.sentimask.SMPoint3F;


/**
 *  This class is for SentiMask Processing Details
 */
public class SMProcessingDetails extends SMBase {

  public SMProcessingDetails() {
    this.SMProcessingDetailsCreate();
  }

  public SMProcessingDetails(long handle) {
    super(handle);
  }

  public native int getObjectSpaceShapePoint3Count();
  public native SMPoint3F getObjectSpaceShapePoint3(int index);

  public native int getImageSpaceShapePoint3Count();
  public native SMPoint3F getImageSpaceShapePoint3(int index);

  public native int getImageSpaceShapePoint2Count();
  public native SMPointF getImageSpaceShapePoint2(int index);

  public native int getLandmarkCount();
  public native SMPointF getLandmark(int index);

  public native double getExpression();
  public native double getPoseParameter();
  public native boolean getFaceDetected();

  private native boolean SMProcessingDetailsCreate();
  
}

File: www/SMProcessingDetails.js
const exec = require('cordova/exec');

const Config = require('./SMConfig');
const SMBase = require('./SMBase');

class SMProcessingDetails extends SMBase {

  getObjectSpaceShapePoint3Count() {
    let point3Count = 0;

    exec((count) => {
      point3Count = count;
    }, (error) => {
      console.error(error);
    }, Config.name, 'SMProcessingDetailsGetObjectSpaceShapePoint3Count', [this.handle]);

    return point3Count;
  }

  getObjectSpaceShapePoint3(index) {
    return new Promise((resolve, reject) => {
      exec(resolve, reject, Config.name, 'SMProcessingDetailsGetObjectSpaceShapePoint3', [this.handle, index]);
    });
  }

  
  getImageSpaceShapePoint3Count() {
    let point3Count = 0;

    exec((count) => {
      point3Count = count;
    }, (error) => {
      console.error(error);
    }, Config.name, 'SMProcessingDetailsGetImageSpaceShapePoint3Count', [this.handle]);

    return point3Count;
  }

  getImageSpaceShapePoint3(index) {
    return new Promise((resolve, reject) => {
      exec(resolve, reject, Config.name, 'SMProcessingDetailsGetImageSpaceShapePoint3', [this.handle, index]);
    });
  }


  getImageSpaceShapePoint2Count() {
    let point2Count = 0;

    exec((count) => {
      point2Count = count;
    }, (error) => {
      console.error(error);
    }, Config.name, 'SMProcessingDetailsGetImageSpaceShapePoint2Count', [this.handle]);

    return point2Count;
  }

  getImageSpaceShapePoint2(index) {
    return new Promise((resolve, reject) => {
      exec(resolve, reject, Config.name, 'SMProcessingDetailsGetImageSpaceShapePoint2', [this.handle, index]);
    });
  }

  getLandmarkCount() {
    let landmarkCount = 0;

    exec((count) => {
      landmarkCount = count;
    }, (error) => {
      console.error(error);
    }, Config.name, 'SMProcessingDetailsGetLandmarkCount', [this.handle]);

    return landmarkCount;
  }

  getLandmark() {
    return new Promise((resolve, reject) => {
      exec(resolve, reject, Config.name, 'SMProcessingDetailsGetLandmark', [this.handle, index]);
    });
  }


  getExpression() {
    let expression = 0;

    exec((pExpression) => {
      expression = pExpression;
    }, (error) => {
      console.error(error);
    }, Config.name, 'SMProcessingDetailsGetExpression', [this.handle]);

    return expression;
  }

  getPoseParameter() {
    let poseParam = 0;

    exec((params) => {
      poseParam = params;
    }, (error) => {
      console.error(error);
    }, Config.name, 'SMProcessingDetailsGetPoseParameter', [this.handle]);

    return poseParam;
  }

  getFaceDetected() {
    let faceDetected = false;

    exec((detected) => {
      faceDetected = detected;
    }, (error) => {
      console.error(error);
    }, Config.name, 'SMProcessingDetailsGetFaceDetected', [this.handle]);

    return faceDetected;
  }

}

module.exports = SMProcessingDetails;
